### PR TITLE
atmega-hal: atmega328pb

### PR DESCRIPTION
**Pull request by explicite**
atmega328pb mcu support like https://github.com/Rahix/avr-hal/commit/356090d155e2ab02599b61a0f0cd2f330c8cd4 and avr-spec

### CODE REVIEW

File: avr-specs/avr-atmega328pb.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+    "arch": "avr",
+    "atomic-cas": false,
+    "cpu": "atmega328pb",
```
Comment (explicite) [inline avr-specs/avr-atmega328pb.json:4]: @Rahix looks like I cannot with this spec. Not sure how to solve this problem

Comment (Rahix) [reply avr-specs/avr-atmega328pb.json:4]: @jkristell, you originally added ATmega328PB support.  What target spec were you compiling with?

Comment (jkristell) [reply avr-specs/avr-atmega328pb.json:4]: I used the atmega328p spec.

Comment (Rahix) [reply avr-specs/avr-atmega328pb.json:4]: Thanks!  @explicite, in that case just drop the custom spec and use `avr-atmega328p.json` in the CI.

**Review by Rahix — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**